front/lib/api/assistant/agent_usage.ts

```diff
@@ -138,47 +139,53 @@ export async function getAgentUsage(
 export async function agentMentionsCount(
   workspaceId: number
 ): Promise<mentionCount[]> {
+  // we are using the front replica because the query is relatively expensive
+  const sequelizeReplica = getFrontReplicaDbConnection();
+
   // We retrieve mentions from conversations in order to optimize the query
   // Since we need to filter out by workspace id, retrieving mentions first
   // would lead to retrieve every single messages
-  const mentions = await Conversation.findAll({
-    attributes: [
-      [
-        Sequelize.literal('"messages->mentions"."agentConfigurationId"'),
-        "agentConfigurationId",
-      ],
-      [
-        Sequelize.fn("COUNT", Sequelize.literal('"messages->mentions"."id"')),
-        "count",
+  const mentions = await sequelizeReplica.transaction(() => {
```

Comment:
This was the only was I found to be able to use the replica without performing a raw sql query

Reply:
Yeah Sequelize is pretty limited, the only "proper" way they offer is when setting up your Sequelize instances but then all the `SELECT` queries will hit the replicas, which is definitely not what we want! 

Let's add a code comment to explain the need for the transaction 🙏